sycl/source/detail/pi.cpp

```diff
@@ -74,18 +78,21 @@ void *loadPlugin(const std::string &PluginPath) {
 // needs to setup infrastructure to route PI_CALLs to the appropriate plugins.
 // Currently, we bind to a singe plugin.
 bool bindPlugin(void *Library) {
-#define STRINGIZE(x) #x
-
-#define _PI_API(api)                                                           \
-  decltype(&api) api##_ptr = ((decltype(&api))(                                \
-      getOsLibraryFuncAddress(Library, STRINGIZE(api##OclPtr))));              \
-  if (!api##_ptr)                                                              \
-    return false;                                                              \
-  api = *api##_ptr;
-#include <CL/sycl/detail/pi.def>
-
-#undef STRINGIZE
-#undef _PI_API
+
+  decltype(::piPluginInit) *PluginInitializeFunction = (decltype(
+      &::piPluginInit))(getOsLibraryFuncAddress(Library, "piPluginInit"));
+  int err = PluginInitializeFunction(&PluginInformation);
```

Comment:
I would implement the init function to not affect the argument if an error occurs. If we need/want that at least the plugin version is set - even on an error - then please document this required behavior.

Though I would prefer to not access the argument if an error is returned in the code below.

Reply:
I have removed piPluginInit function to edit the PluginInformation structure when an error occurs.